### DISCUSSION TITLE
I want to give the start hotkey to quick search

### DISCUSSION THREAD

**Discussion by marvelph** [mikekazakov/nimble-commander#179] [Ideas]
Using modifier keys for quick search can cause conflicts with hotkeys.
Is it possible to introduce a hotkey to start the quick search?

**Comment by mikekazakov** [comment]
That would complicate the logic of key presses routing, which already rather hairy, even further...  
Which hotkeys conflict with e.g. Option modifier?

**Reply by marvelph** [reply]
Thanks for your comment.
I am trying to change all hotkey assignments to match the old console app operation.
I accepted the modification to always use some modifier key because most hotkeys in Nimble Commander require a modifier key.
However, trying to avoid conflicts with Quick Search made it difficult to deal with the need to press many modifier keys at the same time.
If hotkeys did not always require modifier keys, the quick search specification would be fine.

**Reply by mikekazakov** [reply]
But action hotkeys usually have the "Command" modifier, not the "Option" one, aren't they?

**Reply by marvelph** [reply]
That is true, but I avoid it because it causes confusion with the common hotkeys in macOS.

**Reply by mikekazakov** [reply]
Sorry, I still can't understand where the collision comes from, since action hotkeys and quick search can have different modifiers.  
Could you provide a more concrete example?

**Reply by marvelph** [reply]
It is not exactly a conflict, but it consumes a large amount of space available for hotkeys.
For example, if the option key is used for quick search, any hotkey that combines the option key with a letter of the alphabet will be unavailable.
Although there are several modifier keys and they can be used in any combination, the modifier keys that are easy to use are limited because pressing several modifier keys simultaneously is burdensome.
If a hotkey to initiate a quick search were to be introduced, the space consumption available for hotkeys would be minimal.

**Reply by mikekazakov** [reply]
That only counts when considering single-key modifiers, however Quick Search doesn't steal keypresses when other modifiers are present. So, for instance, considering using Option(Alt) for Quick Search, that leaves these different modifiers free for actions hotkeys:
- Command
- Alt+Command
- Ctrl+Command
- Shift+Command
- Ctrl+Alt+Command
- Shift+Alt+Command
- Ctrl
- Ctrl+Shift+Command
- Ctrl+Alt
- Ctrl+Shift
- Ctrl+Shift+Alt
- Shift+Alt  

That should be more than enough to fit any hotkeys unless some very specific action map is needed.

**Reply by marvelph** [reply]
You are correct, and this means that it will be necessary to assign multiple modifier key combinations to hotkeys.
If at all possible, it is preferable that hotkeys not require modifier keys at all.
The old application that I use on a daily basis allows me to perform all operations with hotkeys without modifier keys and with hotkeys using only the shift key.

**Reply by mikekazakov** [reply]
Got it, the desire to map the UX of another application into NC is understandable. Thank you for the input, though I think it would a very niche enchancement that wouldn't benefit most of NC users, so I don't see working on it.

**Reply by marvelph** [reply]
Thanks for discussing this with me.
